controller/proxy-injector/server.go

```diff
@@ -78,6 +78,9 @@ func (w *WebhookServer) Shutdown() error {
 }
 
 func tlsConfig(rootCA *pkgTls.CA, controllerNamespace string) (*tls.Config, error) {
+	// must use the service short name in this TLS identity as the k8s api server
+	// looks for the webhook at <svc_name>.<namespace>.svc, without the cluster
+	// domain.
```

Comment:
Yeah, the suggestion was to use the fully-qualified name here. But the k8s api-server doesn't account for the cluster domain (which makes sense, since cluster domain is user-configurable). I added the comment here, just in case we wonder why we are using the short name here. Happy to remove if we deem the comment isn't necessary.

Reply:
Ah, great, thanks for the additional explanation. Keeping the comment here makes sense to me.